### PR TITLE
Feature/gym 115 create calendar view for workouts

### PR DESCRIPTION
**Pull request by PiotrekBaranowski**
Created Calendar grouping endpoint (Grouping workouts by date), that takes dates as parameters

### CODE REVIEW

**Review by przemyslawren — APPROVED** [review]
LGTM

File: build.gradle [hunk 1]
```diff
@@ -39,6 +39,7 @@ dependencies {
 	implementation 'org.springframework.modulith:spring-modulith-starter-jpa'
 	implementation 'org.springdoc:springdoc-openapi-starter-webmvc-ui:2.6.0'
 	implementation 'org.springframework.boot:spring-boot-starter-actuator:3.4.0'
+	implementation 'org.springframework.boot:spring-boot-starter-webflux'
```
Comment (PiotrekBaranowski) [inline build.gradle:42]: testImplementation

**Comment by PiotrekBaranowski** [comment]
format the code within code styling

**Review by Valentine-456 — APPROVED** [review]
LGTM, I think the tests are ok

**Review by Hubertsci — APPROVED** [review]
Didn't find any faults ;)

File: src/test/java/com/sourcery/gymapp/backend/workout/integration/WorkoutControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        Workout workout4 = WorkoutFactory.createWorkout(
+                userId,
+                "workout4",
+                LocalDateTime.of(2024, 2, 1, 0, 0)
+                        .atZone(ZoneOffset.UTC));
+        Workout workout5 = WorkoutFactory.createWorkout(
+                UUID.randomUUID(),
+                "workout5",
+                LocalDateTime.of(2024, 1, 13, 0, 0)
+                        .atZone(ZoneOffset.UTC));
```
[the remark this answers is not in the document]
Comment (PiotrekBaranowski) [reply src/test/java/com/sourcery/gymapp/backend/workout/integration/WorkoutControllerTest.java:67]: yes, cause this workout has different user UUID, I am testing if it's only fetched for current User

**Pull request merged**